clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1277,8 +1280,7 @@ void prepareShutdown(final Timer timer, final AtomicReference<Throwable> firstEx
 
     // Visible for testing
     void maybeAutoCommitSync(final boolean shouldAutoCommit,
-                             final Timer timer,
-                             final AtomicReference<Throwable> firstException) {
```

Comment:
this parameter wasn't used, and it looks like it is correct to only log the error here.

Reply:
> it looks like it is correct to only log the error here.

Why? Should we not set `firstException` if `null` and `commitSync` throws?

Reply:
It's what the legacy consumer does.

https://github.com/apache/kafka/blob/d066b94c8103cca166d7ea01a4b5bf5f65a3b838/clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java#L1151